scripts/release/tag-images.py

```diff
@@ -0,0 +1,109 @@
+"publish" option it will also publish images to
+https://hub.docker.com/u/vimc
+
+Usage:
+  tag-images.py tag [--publish] <version>
+  tag-images.py publish <version>
+
+"""
+import docker
+import docopt
+from subprocess import run, PIPE
+from release_tag import get_latest_release_tag, validate_release_tag
+
+# List of container images that we will tag
+containers = [
+    "orderly-web",
+    "orderlyweb-migrate",
+    "orderly-web-user-cli",
+    "orderly-web-css-generator"
+]
+
+registry_local = "docker.montagu.dide.ic.ac.uk:5000"
+registry_hub = "vimc"
+
+
+class DockerTag:
+    def __init__(self, registry, name, version=None):
+        self.registry = registry
+        self.name = name
+        self.version = version
+
+    def __str__(self):
+        if self.version:
+            return "{}/{}:{}".format(self.registry, self.name, self.version)
+        else:
+            return "{}/{}".format(self.registry, self.name)
+
+    @property
+    def repository(self):
+        return "{}/{}".format(self.registry, self.name)
+
+    @classmethod
+    def parse(cls, raw):
+        registry, parts = raw.split("/")
+        name, version = parts.split(":")
+        return DockerTag(registry, name, version)
+
+
+def get_version_sha(version):
+    return run(["git", "rev-parse", version],
+               stdout=PIPE, check=True, universal_newlines=True).stdout.strip()
+
+
+def set_image_tag(name, version, sha):
+    d = docker.client.from_env()
+    img = d.images.pull(str(DockerTag(registry_local, name, sha)))
+    tag_and_push(img, registry_local, name, version)
+
+
+def set_image_tags(version):
+    sha = get_version_sha(version)
+    print("Setting image tags")
+    for name in containers:
+        print("  - " + name)
+        set_image_tag(name, version, sha)
+
+
+def publish_images(version):
+    d = docker.client.from_env()
+    print("Pushing release to docker hub")
+    for name in containers:
+        img = d.images.get(str(DockerTag(registry_local, name, version)))
+        publish_image(img, name)
+
+
+def publish_image(img, name):
+    tags = [DockerTag.parse(x) for x in img.tags]
+    published = [t.version for t in tags if t.registry == registry_hub]
+    existing = [t.version for t in tags if t.registry == registry_local]
+    for tag in set(existing) - set(published):
+        tag_and_push(img, registry_hub, name, tag)
+
+
+# NOTE: Using subprocess here and not the python docker module because
+# the latter does not support streaming as nicely as the CLI
+def tag_and_push(img, registry_local, name, tag):
+    t = DockerTag(registry_local, name, tag)
+    img.tag(t.repository, t.version)
+    run(["docker", "push", str(t)], check=True)
+
+
+if __name__ == "__main__":
+    args = docopt.docopt(__doc__)
+    version = args["<version>"]
+    if version == "latest":
+        version = get_latest_release_tag()
+    else:
+        validate_release_tag(version)
+    if args["tag"]:
```

Comment:
Do we ever use this in any other way than `tag --publish`? If so maybe we roll these into a single command? Can you think of a situation in which we'd want to tag and not publish?

Reply:
Will replace with --local-only flag so publish is the default